_scripts/migrations/0015_discussion_access_and_invites.sql

```diff
@@ -0,0 +1,98 @@
+ALTER TABLE discussions ADD COLUMN IF NOT EXISTS public_access boolean default true not null;
+
+CREATE TABLE IF NOT EXISTS discussion_flair_access (
+    discussion_id varchar(36) not null,
+    flair_template_id varchar(36) not null,
+    created_at timestamp with time zone default current_timestamp not null,
+    deleted_at timestamp with time zone,
+    PRIMARY KEY(discussion_id, flair_template_id)
+);
+
+CREATE TABLE IF NOT EXISTS discussion_user_access (
+    discussion_id varchar(36) not null,
+    user_id varchar(36) not null,
+    created_at timestamp with time zone default current_timestamp not null,
+    deleted_at timestamp with time zone,
+    PRIMARY KEY(user_id, discussion_id)
+);
+
+CREATE TABLE IF NOT EXISTS discussion_invite_link_access (
+    discussion_id varchar(36) PRIMARY KEY,
+    vip_invite_url text,
+    invite_url text,
+    created_at timestamp with time zone default current_timestamp not null,
+    updated_at timestamp with time zone default current_timestamp not null,
+    deleted_at timestamp with time zone
+);
+
+CREATE TABLE IF NOT EXISTS discussion_user_invitations (
+    id varchar(36) PRIMARY KEY,
+    user_id varchar(36) not null,
+    discussion_id varchar(36) not null,
+    invite_from_user_id varchar(36) not null,
+    created_at timestamp with time zone default current_timestamp not null,
+    updated_at timestamp with time zone default current_timestamp not null,
+    deleted_at timestamp with time zone,
+    status varchar(20) not null
+);
+
+CREATE TABLE IF NOT EXISTS discussion_user_requests (
+    id varchar(36) PRIMARY KEY,
+    user_id varchar(36) not null,
+    discussion_id varchar(36) not null,
+    created_at timestamp with time zone default current_timestamp not null,
+    updated_at timestamp with time zone default current_timestamp not null,
+    deleted_at timestamp with time zone,
+    status varchar(20) not null
+);
+
+ALTER TABLE discussion_flair_access
+    ADD CONSTRAINT dfa_discussions_fk_79064bd041e0 FOREIGN KEY (discussion_id) REFERENCES discussions(id) MATCH FULL,
```

Comment:
I think all of these that reference a discussion should cascade on delete because otherwise we won't be able to delete discussions, right?